### PR TITLE
Grant access to support group during WifConfig creation

### PR DESCRIPTION
**Pull request by renan-campos**
This enhancement represents the frontend portion of OCM-10387. 

The set of changes included enhances the creation logic by binding the roles to the principal used for servicing GCP projects. The principal and roles principal are specified by the WifConfig object received from the backend creation call. When run in manual mode, the equivalent gcloud commands are also provided.

A refactor was done to provide this new functionality. Helpers that modified GCP resources based on the returned WifConfig were encapsulated by the GcpClientWifConfigShim object. This was done to simplify the signature of methods called by the create command and to improve the reusability of the methods shared by multiple operations.

The GCP client package was also refactored. To fit convention, all methods take a context as their first parameter. Unused methods were removed, and the remaining methods were alphabetized.

### CODE REVIEW

File: cmd/ocm/gcp/scripting.go [hunk 1]
```diff
@@ -186,6 +188,40 @@ func createServiceAccountScriptContent(wifConfig *cmv1.WifConfig, projectNum int
 	return sb.String()
 }
 
+func grantSupportAccessScriptContent(wifConfig *cmv1.WifConfig) string {
```
[the remark this answers is not in the document]
Comment (renan-campos) [reply cmd/ocm/gcp/scripting.go:191]: I'll update this and attach an example script.

**Comment by renan-campos** [comment]
[scripts.zip](https://github.com/user-attachments/files/16932767/scripts.zip)

**Comment by renan-campos** [comment]
marking pr as draft until backend enhancements have been merged

**Review by ckandag — APPROVED** [review]

**Review by JakobGray — COMMENTED** [review]
I'm good with the changes as they are. I think I'd like to iterate on the refactoring further in other PRs.

**Pull request merged**